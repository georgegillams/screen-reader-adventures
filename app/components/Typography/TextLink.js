import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import BpkSmallNewWindowIcon from 'bpk-component-icon/sm/new-window';

import STYLES from './typography.scss';
import { cssModules } from 'bpk-react-utils';
const getClassName = cssModules(STYLES); // REGEX_REPLACED

const TextLink = props => {
  const {
    external,
    fancy,
    light,
    href,
    className,
    textClassName,
    children,
    onClick,
    ...rest
  } = props;

  const classNameFinal = [getClassName('typography__main')];
  classNameFinal.push(getClassName('typography__link'));
  classNameFinal.push(getClassName('typography__link--text-link'));
  classNameFinal.push(getClassName('typography--no-padding'));
  if (light) {
    classNameFinal.push(getClassName('typography--light'));
    classNameFinal.push(getClassName('typography--light--text-link'));
  }
  if (fancy) classNameFinal.push(getClassName('typography--fancy'));
  classNameFinal.push(getClassName('typography--inline'));
  if (className) {
    classNameFinal.push(className);
  }

  return external ? (
    <a
      href={href}
      rel="noopener noreferrer"
      target="_blank"
      className={classNameFinal.join(' ')}
      onClick={onClick}
      {...rest}
    >
      {children}
      <div className={getClassName('typography__icon')}>
        <BpkSmallNewWindowIcon
          className={getClassName('typography__icon--inner')}
        />
      </div>
    </a>
  ) : (
    <Link
      to={href}
      onClick={onClick}
      className={classNameFinal.join(' ')}
      {...rest}
    >
      {children}
    </Link>
  );
};

TextLink.propTypes = {
  onClick: PropTypes.func,
  fancy: PropTypes.bool,
  external: PropTypes.bool,
  light: PropTypes.bool,
  children: PropTypes.node,
  href: PropTypes.string,
  textClassName: PropTypes.string,
  className: PropTypes.string,
};

TextLink.defaultProps = {
  onClick: null,
  external: false,
  fancy: false,
  light: false,
  href: null,
  children: null,
  textClassName: null,
  className: null,
};

export default TextLink;
