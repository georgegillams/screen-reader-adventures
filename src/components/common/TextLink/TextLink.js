import React from 'react';
import PropTypes from 'prop-types';
import { TextLink as GGTextLink } from 'gg-components/TextLink';
import Link from 'next/link';
import nextifyHref from 'client-utils/nextifyHref';

const TextLink = props => {
  const { href, hrefExternal, ...rest } = props;
  if (hrefExternal) {
    return <GGTextLink {...props} />;
  }

  const destination = nextifyHref(href);
  return (
    <Link passHref href={destination.url} as={destination.as} {...destination.options}>
      <GGTextLink href={href} {...rest} />
    </Link>
  );
};

TextLink.propTypes = {
  href: PropTypes.string.isRequired,
  hrefExternal: PropTypes.bool,
};
TextLink.defaultProps = {
  hrefExternal: false,
};

export default TextLink;
