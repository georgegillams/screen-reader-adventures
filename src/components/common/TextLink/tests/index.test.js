import React from 'react';

import { render } from '@testing-library/react';

import TextLink from '..';

describe('<TextLink />', () => {
  const children = 'some text';

  it('Should render correctly', () => {
    const { container } = render(<TextLink href="/test">{children}</TextLink>);

    expect(container).toMatchSnapshot();
  });

  it('Should render with nextified href', () => {
    const { container } = render(<TextLink href="/blog/test">{children}</TextLink>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(
      <TextLink href="/test" className={'test-custom-classname'}>
        {children}
      </TextLink>
    );

    expect(container).toMatchSnapshot();
  });
  // This test was causing jest to hang. Reinstate if possible
  // it('Should render correctly with hrefExternal', () => {
  //   const { container } = render(
  //     <TextLink href="/test" hrefExternal>
  //       {children}
  //     </TextLink>
  //   );

  //   expect(container).toMatchSnapshot();
  // });
});
