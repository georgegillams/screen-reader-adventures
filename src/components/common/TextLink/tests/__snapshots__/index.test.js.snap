// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> Should render correctly 1`] = `
<div>
  <a
    class="text-link text-link--text-link text-link--no-padding text-link--inline"
    href="/test"
  >
    some text
  </a>
</div>
`;

exports[`<TextLink /> Should render correctly with custom className 1`] = `
<div>
  <a
    class="text-link text-link--text-link text-link--no-padding text-link--inline test-custom-classname"
    href="/test"
  >
    some text
  </a>
</div>
`;

exports[`<TextLink /> Should render with nextified href 1`] = `
<div>
  <a
    class="text-link text-link--text-link text-link--no-padding text-link--inline"
    href="/blog/test"
  >
    some text
  </a>
</div>
`;
